### PR TITLE
Add error check for pressure pin in the NSFVAction.

### LINKED ISSUE #21617: NSFVAction does not complain if pressure pin parameters are not set by the user.
## Bug Description
The NSFVAction does not complain if pressure_pin related parameters are not set by the user when the functionality is enabled.

## Steps to Reproduce
Run `modules/navier_stokes/test/test/finite_volume/ins/lid_driven/lid_driven-action.i` without `pressure_pin_type` defined.
It runs, while it would be nicer if it warned the user that a default value is chosen. 

## Impact
May lead to different constraints compared to what was intended.

### PR DESCRIPTION
**Pull request by grmnptr**
Closes #21617

### CODE REVIEW

**Comment by grmnptr** [comment]
I also changed some solver options in the NS module. I had format-on-save turned on with moose formatting so it changed a few lines.

File: modules/navier_stokes/src/actions/NSFVAction.C [hunk 1]
```diff
@@ -2571,13 +2580,19 @@ NSFVAction::checkPassiveScalarParameterErrors()
...
+  if (should_be_defined)
+    not_word = "";
+
   for (const auto & param : dependent_parameters)
-    if (_pars.isParamSetByUser(param))
+    if (_pars.isParamSetByUser(param) == !should_be_defined)
       paramError(param,
-                 "This parameter should not be given by the user with the corresponding " +
-                     main_parameter + " setting!");
+                 "This parameter should " + not_word +
```
Comment (lindsayad) [inline modules/navier_stokes/src/actions/NSFVAction.C:2593]:
```suggestion
                 "This parameter should " + (should_be_defined ? "" : "not") +
```

File: modules/navier_stokes/src/actions/NSFVAction.C [hunk 2]
```diff
@@ -2571,13 +2580,19 @@ NSFVAction::checkPassiveScalarParameterErrors()
...
 void
 NSFVAction::checkDependentParameterError(const std::string main_parameter,
-                                         const std::vector<std::string> dependent_parameters)
+                                         const std::vector<std::string> dependent_parameters,
+                                         const bool should_be_defined)
 {
+  std::string not_word = "not";
+  if (should_be_defined)
+    not_word = "";
+
```
Comment (lindsayad) [inline modules/navier_stokes/src/actions/NSFVAction.C:2589]:
```suggestion
```

**Review by lindsayad — CHANGES_REQUESTED** [review]
Are most of the input file changes automated formatting from VSCode?

**Comment by grmnptr** [comment]
Yes, those are changes from applying the moose-format on save in vscode. I can roll them back if you think I should!

**Comment by lindsayad** [comment]
I like the changes, but in the future it might be nice to separate commits, e.g. one commit "formatting changes" and then the next "change to LU". I wouldn't go through the effort of splitting what you've already done here though. That would probably take a while.

File: modules/navier_stokes/src/actions/NSFVAction.C [hunk 3]
```diff
@@ -2571,13 +2580,15 @@ NSFVAction::checkPassiveScalarParameterErrors()
...
+                                         const std::vector<std::string> dependent_parameters,
+                                         const bool should_be_defined)
 {
   for (const auto & param : dependent_parameters)
-    if (_pars.isParamSetByUser(param))
+    if (_pars.isParamSetByUser(param) == !should_be_defined)
       paramError(param,
-                 "This parameter should not be given by the user with the corresponding " +
-                     main_parameter + " setting!");
+                 "This parameter should " + std::string(should_be_defined ? "" : "not") +
```
Comment (lindsayad) [inline modules/navier_stokes/src/actions/NSFVAction.C:2589]: Did the compiler require explicitly typing this?

File: modules/navier_stokes/test/tests/finite_volume/ins/block_restriction/2d-rc.i [hunk 4]
```diff
@@ -252,9 +252,8 @@ restricted_blocks = '1'
 [Executioner]
   type = Steady
   solve_type = 'NEWTON'
-  petsc_options_iname = '-pc_type -ksp_gmres_restart -sub_pc_type -sub_pc_factor_shift_type'
-  petsc_options_value = 'asm      100                lu           NONZERO'
-  line_search = 'none'
+  petsc_options_iname = '-pc_type -pc_factor_shift_type'
+  petsc_options_value = 'lu NONZERO'
```
Comment (lindsayad) [inline modules/navier_stokes/test/tests/finite_volume/ins/block_restriction/2d-rc.i:256]: With only two options this is fine as-is, but in general I like to space the values to align with the names to help readability.

**Review by lindsayad — APPROVED** [review]

File: modules/navier_stokes/test/tests/finite_volume/ins/block_restriction/2d-rc.i (hunk 4 above)
Comment (grmnptr) [reply modules/navier_stokes/test/tests/finite_volume/ins/block_restriction/2d-rc.i:256]: Got it, will pay attention to it!

File: modules/navier_stokes/src/actions/NSFVAction.C (hunk 3 above)
Comment (grmnptr) [reply modules/navier_stokes/src/actions/NSFVAction.C:2589]: Yes, conversion from char[13] caused compile-time-errors.

**Comment by cticenhour** [comment]
Seeing intermittent failures on CIVET for the `navier_stokes/test:finite_volume/pins/mms.rc-2d` test which might be the result of this set of changes. An example is here: https://civet.inl.gov/job/1117452/

**Comment by grmnptr** [comment]
Thanks! Does it depend on the machine it is running on?

**Comment by grmnptr** [comment]
I'll fix this tonight.

**Comment by cticenhour** [comment]
It seems to only be triggering on the Linux test boxes.

**Comment by grmnptr** [comment]
Yes, it is caused by my change and the reason it changes from machine to machine is because they have different amount of memory. I'll fix this in a moment.

**Comment by grmnptr** [comment]
I hope this fixes it:
https://github.com/idaholab/moose/pull/21659

Is there a way to test it on the machine with the lowest memory?

**Comment by cticenhour** [comment]
With enough permission you should be able to invalidate and choose the exact machine to test on, but I don't know which one would have the least memory. @milljm @loganharbour which one would that be?

**Comment by cticenhour** [comment]
You might just pick one of the failed recipes and select the machine that was running for that case.

**Comment by loganharbour** [comment]
they all have 128G - and historically 75% of is that is pretty much always available. If you're running something with with that big of a memory requirement I suggest not making it a test because that's way too much.

**Pull request merged**